### PR TITLE
Add 'Firebolt' to Prefect task library.

### PR DESCRIPTION
**Pull request by raghavSharmaSigmoid**
## Summary
This PR is raised to add 'Firebolt' to the Prefect task library.

## Changes
Addition of tasks and tests files of 'Firebolt'.

## Importance
After merging, users will be able to generate ETL flows using 'Firebolt' tasks.

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

**Comment by raghavSharmaSigmoid** [comment]
@tpdorsey @madkinsz @desertaxle 
Request you to review this PR. Thanks!

File: tests/tasks/firebolt/test_firebolt.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        Tests that all required params are present for FireboltQuery Task.
+        """
+        task = FireboltQuery(
+            database="test",
+            username="test",
+            password="test",
+            engine_name="test",
+            query="test",
+        )
+        assert task.database is not None
```
Comment (desertaxle) [inline tests/tasks/firebolt/test_firebolt.py:38]:
```suggestion
        assert task.database == "test"
        assert task.username == "test"
        assert task.password == "test"
        assert task.engine_name == "test"
        assert task.query == "test"
```

File: tests/tasks/firebolt/test_firebolt.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        mock_conn.return_value.__enter__.return_value.execute.return_value = 0
+        query = "SHOW DATABASES"
+        output = FireboltQuery(
+            database="test",
+            username="test",
+            password="test",
+            engine_name="test",
+            query=query,
+        ).run()
+        mock_conn.assert_called_with()
```
Comment (desertaxle) [inline tests/tasks/firebolt/test_firebolt.py:105]: I think this assertion should check to make sure that the `connect` method was called with the expected arguments (`database`, `username`, etc.)

File: tests/tasks/firebolt/test_firebolt.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        mock_conn.return_value.__enter__.return_value.fetchall.return_value = ["TESTDB"]
+        query = "SHOW DATABASES"
+        output = FireboltQuery(
+            database="test",
+            username="test",
+            password="test",
+            engine_name="test",
+            query=query,
+        ).run()
+        mock_conn.assert_called_with()
```
Comment (desertaxle) [inline tests/tasks/firebolt/test_firebolt.py:129]: I think this assertion should check to make sure that the `connect` method was called with the expected arguments (`database`, `username`, etc.)

**Review by desertaxle — CHANGES_REQUESTED** [review]
Thanks @raghavSharmaSigmoid! Overall this looks good. I have a few comments on the tests. Let me know if any comments require further clarification.

File: src/prefect/tasks/firebolt/__init__.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+"""
+This module contains a collection of tasks for interacting with firebolt databases via
```
Comment (tpdorsey) [inline src/prefect/tasks/firebolt/__init__.py:2]:
```suggestion
This module contains a collection of tasks for interacting with Firebolt databases via
```

File: src/prefect/tasks/firebolt/__init__.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+the firebolt-sdk-python library.
```
Comment (tpdorsey) [inline src/prefect/tasks/firebolt/__init__.py:3]: I think this should be "firebolt-python-sdk" per https://github.com/firebolt-db/firebolt-python-sdk

File: src/prefect/tasks/firebolt/firebolt.py [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+        Args:
+            - database (str): name of the database to use.
+            - username (str): username used to authenticate.
+            - password (str): password used to authenticate.
+            - engine_name (str): name of the engine to use.
+            - query (str): query to execute against database.
+
+        Returns:
+            - List[List]: output of cursor.fetchall() if cursor.rowcount > 0 else an empty list
```
Comment (tpdorsey) [inline src/prefect/tasks/firebolt/firebolt.py:57]:
```suggestion
            - List[List]: output of `cursor.fetchall()` if `cursor.rowcount` > 0, else an empty list.
```

File: src/prefect/tasks/firebolt/firebolt.py [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+        Raises:
+            - ValueError: if a required parameter is not supplied
```
Comment (tpdorsey) [inline src/prefect/tasks/firebolt/firebolt.py:60]:
```suggestion
            - ValueError: if a required parameter is not supplied.
```

File: src/prefect/tasks/firebolt/firebolt.py [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+            - DatabaseError: if exception occurs when executing the query
```
Comment (tpdorsey) [inline src/prefect/tasks/firebolt/firebolt.py:61]:
```suggestion
            - DatabaseError: if exception occurs when executing the query.
```

**Review by tpdorsey — CHANGES_REQUESTED** [review]
Docstrings look good pending my suggested changes. Thanks!

**Comment by raghavSharmaSigmoid** [comment]
@desertaxle @tpdorsey  have refactored the code and updated docstrings as per your review comments. Please take a look. Thanks!

**Review by tpdorsey — DISMISSED** [review]

**Review by desertaxle — APPROVED** [review]

**Comment by raghavSharmaSigmoid** [comment]
Thanks for approving the changes @desertaxle @tpdorsey.
@madkinsz request you to review this PR for merging.

File: tests/tasks/firebolt/test_firebolt.py [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+    # test to check if the query was executed and metadata was retrieved from database
+    def test_execute_fetchall(self, mock_conn):
+        """
+        Tests that the FireboltQuery Task calls the fetchall method on the cursor.
+        """
+        connection = mock_conn[0]
+        cursor = mock_conn[1]
+
+        # setting execute return
+        cursor.return_value.__enter__.return_value.execute.return_value = 1
```
Comment (madkinsz) [inline tests/tasks/firebolt/test_firebolt.py:134]: I generally have a preference for something like
```suggestion
        cursor().__enter__().execute.return_value = 1
```
as it's easier to read. No reason for it to block this though.

Comment (raghavSharmaSigmoid) [reply tests/tasks/firebolt/test_firebolt.py]: @madkinsz thanks for reviewing. Have made the suggested changes. Please verify.

**Review by madkinsz — APPROVED** [review]

**Comment by raghavSharmaSigmoid** [comment]
Thankyou for reviewing, approving and merging this pull request.

**Pull request merged**